src/aequitas/flow/methods/preprocessing/label_flipping.py

```diff
@@ -0,0 +1,138 @@
+from .preprocessing import PreProcessing
+
+from ...utils import create_logger
+from ...utils.imports import import_object
+
+import inspect
+import pandas as pd
+from typing import Optional, Tuple
+import numpy as np
+from sklearn.ensemble import BaggingClassifier
+
+METHODS = ["ensemble_margin", "residuals"]
+
+class LabelFlipping(PreProcessing):
+    def __init__(
+            self,
+            flip_rate: float = 0.1,
+            bagging_max_samples: float = 0.5,
+            base_estimator: str = "sklearn.tree.DecisionTreeClassifier",
+            base_estimator_args: dict = {}, 
+            n_estimators: int = 10,
+            fair_ordering: bool = True,
+            ordering_method: str = "ensemble_margin",
```

Comment:
One tip, for these types of variables with a fixed number of possible values, you can use literals:  https://peps.python.org/pep-0586/

I have not used it for everything tbh, but it might be useful!